system/BaseModel.php

```diff
@@ -0,0 +1,1825 @@
+<?php
+
+/**
+ * This file is part of the CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace CodeIgniter;
+
+use Closure;
+use CodeIgniter\Database\BaseResult;
+use CodeIgniter\Database\Exceptions\DatabaseException;
+use CodeIgniter\Database\Exceptions\DataException;
+use CodeIgniter\Exceptions\ModelException;
+use CodeIgniter\I18n\Time;
+use CodeIgniter\Pager\Pager;
+use CodeIgniter\Validation\ValidationInterface;
+use Config\Services;
+use ReflectionClass;
+use ReflectionException;
+use ReflectionProperty;
+use stdClass;
+
+/**
+ * Class Model
+ *
+ * The Model class provides a number of convenient features that
+ * makes working with a database table less painful.
+ *
+ * It will:
+ *      - automatically connect to database
+ *      - allow intermingling calls between db connection, the builder,
+ *          and methods in this class.
+ *      - simplifies pagination
+ *      - removes the need to use Result object directly in most cases
+ *      - allow specifying the return type (array, object, etc) with each call
+ *      - ensure validation is run against objects when saving items
```

Comment:
I think this docblock needs to be revised since this `BaseModel` is DB-agnostic?

Reply:
Sure. A lot of comments have to be reworked.
 
I also noticed something else. 
$this->table is not used anywhere in BaseModel now so it should be moved to Model as well. And remove $this->table from all the comments

Also with all this. I'm noticing now how simple it will be to add support for Composite Keys. I'm not sure if this is to be implemented with this PR or should I make additional one for such purpose 

```
	protected $primaryKey = ['composite1','composite2'];
```

Then find can be called like this

```
      $model->find(['composite1_value', 'composite2_value'])
```

Reply:
Composite primary keys should be for another PR and discussion. 👍